Dalamud/Configuration/Internal/DalamudConfiguration.cs

```diff
@@ -508,7 +514,8 @@ private void Save()
     {
         ThreadSafety.AssertMainThread();
 
-        Service<ReliableFileStorage>.Get().WriteAllText(
+        this.writeTask?.Wait();
+        this.writeTask = Service<ReliableFileStorage>.Get().WriteAllTextAsync(
             this.configPath, JsonConvert.SerializeObject(this, SerializerSettings));
         this.DalamudConfigurationSaved?.Invoke(this);
```

Comment:
We'll be invoking `DalamudConfigurationSaved` before the save actually completes. This *seems* fine, given that the only use I can see is updating `DalamudPluginInterface#GeneralChatType`, but it does feel like a behavioral change that should at least be noted.